### PR TITLE
rangefeed: extract adaptor interface for underlying KV

### PR DESCRIPTION
**Pull request by irfansharif**
This helps break a dependency link between pkg/.../rangefeed and
pkg/.../kvcoord, something we want in #73876 to enable a low-level
testing package to create rangefeeds. This commit is an alternative to
\#74865, which does the same by carving out package external tests
instead.

Release note: None

### CODE REVIEW

File: pkg/kv/kvclient/rangefeed/rangefeed.go [hunk 1]
```diff
@@ -41,8 +39,8 @@ import (
 // TODO(yevgeniy): Instead of rolling our own logic to parallelize scans, we should
 // use streamer API instead (https://github.com/cockroachdb/cockroach/pull/68430)
 
-// kvDB is an adapter to the underlying KV store.
-type kvDB interface {
+// KVDB is an adapter to the underlying KV store.
+type KVDB interface {
```
Comment (ajwerner) [inline pkg/kv/kvclient/rangefeed/rangefeed.go:43]: I think DB would be fine

File: pkg/kv/kvclient/rangefeed/config.go [hunk 2]
```diff
@@ -152,31 +154,39 @@ func initConfig(c *config, options []Option) {
...
-	if c.targetScanBytes == 0 {
-		c.targetScanBytes = 1 << 19 // 512 KiB
+	if c.TargetScanBytes == 0 {
+		c.TargetScanBytes = 1 << 19 // 512 KiB
 	}
 }
 
+// TestingScanConfigWithOptions returns a ScanConfig for tests initialized using
+// the provided options.
+func TestingScanConfigWithOptions(options []Option) ScanConfig {
```
Comment (ajwerner) [inline pkg/kv/kvclient/rangefeed/config.go:164]: Feels like this should be variadic

File: pkg/kv/kvclient/rangefeed/config.go [hunk 3]
```diff
@@ -37,24 +37,26 @@ type config struct {
...
 }
 
-type scanConfig struct {
-	// scanParallelism controls the number of concurrent scan requests
+// ScanConfig is used to configure rangefeed scans.
+type ScanConfig struct {
+	// ScanParallelism controls the number of concurrent scan requests
 	// that can be issued.  If unspecified, only 1 scan request at a time is issued.
-	scanParallelism func() int
+	ScanParallelism func() int
```
Comment (ajwerner) [inline pkg/kv/kvclient/rangefeed/config.go:44]: It sorta feels like if we're going to stick with the functional options then we should expose the values via methods.

**Review by ajwerner — COMMENTED** [review]
Thanks for doing this. I wonder if we need to step back on the goal in terms of dependency untangling. I don't think this pr is a mistake, but I'm not totally in love with it either if it doesn't solve the dependency problem on its own. It feels like we need to inject a constructor for this db adapter which the testcluster can construct without actually importing our new package.

**Comment by irfansharif** [comment]
Should we just stick to https://github.com/cockroachdb/cockroach/pull/74865 then?

**Comment by ajwerner** [comment]
Potentially. I want somebody on kv who's thought about the dependency design more than me to weigh in. I won't hold up the spanconfig stuff on any of this.

**Comment by irfansharif** [comment]
I'll drop this commit from #73876 for now then.

**Comment by irfansharif** [comment]
I think I'm just going to drop this PR, we don't really need it for anything and I'm not sure if it's even a good refactor.

**Pull request closed without merging**